packages/panna-sdk/src/core/utils/activity.ts

```diff
@@ -0,0 +1,454 @@
+              tokenInfo: {
+                address: erc721Tx?.token.address,
+                name: erc721Tx?.token.name,
+                symbol: erc721Tx?.token.symbol,
+                decimals: erc721Tx?.token.decimals,
+                type: TokenERC.ERC721,
+                icon: erc721Tx?.token.icon_url
+              }
+            }
+          } as unknown as ERC721Amount;
+          break;
+        }
+
+        case TokenERC.ERC1155: {
+          const erc1155Tx = tx.token_transfers.find(
+            (e) => e.token.type.toLowerCase() === TokenERC.ERC1155
+          );
+          const erc1155TxTotal = erc1155Tx?.total as BlockscoutTotalERC1155;
+
+          amount = {
+            type: TokenERC.ERC1155,
+            tokenId: erc1155TxTotal.token_id,
+            value: erc1155TxTotal.value,
+            instance: {
+              id: erc1155TxTotal.token_instance.id,
+              isUnique: erc1155TxTotal.token_instance.is_unique,
+              owner: erc1155TxTotal.token_instance.owner,
+              tokenInfo: {
+                address: erc1155Tx?.token.address,
+                name: erc1155Tx?.token.name,
+                symbol: erc1155Tx?.token.symbol,
+                decimals: erc1155Tx?.token.decimals,
+                type: TokenERC.ERC1155,
+                icon: erc1155Tx?.token.icon_url
+              }
+            }
+          } as unknown as ERC1155Amount;
+          break;
+        }
+      }
+
+      const activity: Activity = {
+        activityType,
+        transactionID,
+        amount,
+        status
+      };
+      return activity;
+    });
+
+  const metadata: ActivityMetadata = {
+    count: activities.length,
+    offset,
+    limit
+  };
+
+  const result: GetActivityResult = { activities, metadata };
+
+  return result;
+};
+
+export const fillTokenTransactions = async (
+  address: string,
+  transactions: BlockscoutTransaction[],
+  recursionTxHash?: string
+): Promise<BlockscoutTransaction[]> => {
+  const isRecursion: boolean = !!recursionTxHash;
+
+  const cacheKeyTokenTransferTxs = `${address}_tt`;
+  const cacheKeyTokenTransferNextPageParams = `${address}_tt_params`;
+
+  let userTokenTransfers: BlockscoutTokenTransfer[] = activityCache.has(
+    cacheKeyTokenTransferTxs
+  )
+    ? (activityCache.get(cacheKeyTokenTransferTxs) as BlockscoutTokenTransfer[])
+    : [];
+
+  let nextPageParams: BlockscoutNextPageParams | null = activityCache.has(
+    cacheKeyTokenTransferNextPageParams
+  )
+    ? (activityCache.get(
+        cacheKeyTokenTransferNextPageParams
+      ) as BlockscoutNextPageParams)
+    : null;
+
+  if (isRecursion && nextPageParams === null) {
+    // On recurred invocation, when no more pages (token-transfers) are available to check through,
+    // return all the transactions in their current state
+    return transactions;
+  }
+
+  const baseRequestUrl = `https://blockscout.lisk.com/api/v2/addresses/${address}/token-transfers`;
```

Comment:
Optional: I think we can create a small funcs, in future it would be easier to manage all blockscout endpoints if we use more of them
```ts
const BASE_BLOCKSCOUT_URL = 'https://blockscout.lisk.com/api/v2'

const getBaseTokenTransferRequestUrl(address) => 
`${BASE_BLOCKSCOUT_URL}/addresses/${address}/token-transfers`

const getBaseTransactionsRequestUrl(address) => 
`${BASE_BLOCKSCOUT_URL}/addresses/${address}/transactions`
```